### PR TITLE
Waypoint doc updates

### PR DESCRIPTION
**Pull request by craigbox**
Update the waypoint documentation:

- move policy and debugging to their own pages
- clarify L4 vs L7 policy
- reorder a little

### CODE REVIEW

File: content/en/docs/ambient/usage/waypoint/index.md [hunk 1]
```diff
@@ -10,25 +10,30 @@ test: no
...
+When you add a waypoint proxy for a workload, traffic will be forwarded from the ztunnel to that waypoint. If your applications require any of the following L7 mesh functions, you will need to use a waypoint proxy:
 
 * **Traffic management**: HTTP routing & load balancing, circuit breaking, rate limiting, fault injection, retries, timeouts
 * **Security**: Rich authorization policies based on L7 primitives such as request type or HTTP header
 * **Observability**: HTTP metrics, access logging, tracing
 
 ## Deploy a waypoint proxy
 
-Waypoint proxies are deployed declaratively using Kubernetes Gateway resources or the helpful istioctl command. By default, a waypoint will only handle `service` traffic. It is possible for the bundled Istio waypoint class to handle `workload` traffic. Istio also recognizes a convenient `all` value which includes both previously mentioned traffic types. Finally, a waypoint may be configured for testing to handle no traffic at all by using the `none` value.
+Waypoint proxies are deployed declaratively using Kubernetes Gateway resources. The istioctl command can be used to generate a manifest for a waypoint.
```
Comment (ilrudie) [inline content/en/docs/ambient/usage/waypoint/index.md:34]:
```suggestion
Waypoint proxies are deployed declaratively using Kubernetes Gateway resources. You can use istioctl experimental subcommands to generate, apply, and list these resoures.
```

File: content/en/docs/ambient/usage/waypoint/index.md [hunk 2]
```diff
@@ -140,14 +132,18 @@ $ kubectl label ns default istio.io/use-waypoint=waypoint
 namespace/default labeled
 {{< /text >}}
 
-After a namespace is enrolled to use a waypoint, any requests from any pods in ambient to any service running in that namespace will be routed through the waypoint for L7 processing and policy enforcement.
+After a namespace is enrolled to use a waypoint, any requests from any pods enrolled in the mesh to any service running in that namespace will be routed through the waypoint for L7 processing and policy enforcement.
```
Comment (ilrudie) [inline content/en/docs/ambient/usage/waypoint/index.md:135]: This is kind of misleading. Sidecar-enabled pods are part of the mesh but they won't use the waypoint...

**Review by ilrudie — COMMENTED** [review]
Nice! I left a number of suggestions but some may be out of scope for what you wish to accomplish here.  If so I can try a fast follower to get some of them knocked out.

File: content/en/docs/ambient/usage/debug-waypoint/index.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+title: Debug connectivity issues with waypoints
+description: How to investigate problems routing through waypoint proxies.
+weight: 70
+owner: istio/wg-networking-maintainers
+test: no
+---
+
+This guide describes what to do if you have enrolled a workload in a waypoint proxy, but you are not seeing the expected behavior.
+
+## Problems with authentication policy
```
Comment (linsun) [inline content/en/docs/ambient/usage/debug-waypoint/index.md:11]: The guide is not only written for authn policy.  maybe reword it to L7 policy not enforced/processed by waypoint?

File: content/en/docs/ambient/usage/add-workloads/index.md [hunk 4]
```diff
@@ -62,7 +66,7 @@ The following labels control if a resource is included in the mesh in ambient mo
 |  Name  | Feature Status | Resource | Description |
 | --- | --- | --- | --- |
 | `istio.io/dataplane-mode` | Beta | `Namespace` or `Pod` (latter has precedence) |  Add your resource to an ambient mesh. <br><br> Valid values: `ambient` or `none`. |
-| `istio.io/use-waypoint` | Beta | `Namespace`, `Service` or `Pod` | Use a waypoint for traffic to the labeled resource for L7 policy enforcement. <br><br> Valid values: `{waypoint-name}`, `{namespace}/{waypoint-name}`, or `#none` (with hash). |
+| `istio.io/use-waypoint` | Beta | `Namespace`, `Service` or `Pod` | Use a waypoint for traffic to the labeled resource for L7 policy enforcement. <br><br> Valid values: `{waypoint-name}`, `{namespace}/{waypoint-name}`, or `none`. |
```
[the remark this answers is not in the document]
Comment (linsun) [reply content/en/docs/ambient/usage/add-workloads/index.md:69]: that is a good catch Ian

File: content/en/docs/ambient/usage/add-workloads/index.md [hunk 5]
```diff
@@ -62,7 +66,7 @@ The following labels control if a resource is included in the mesh in ambient mo
...
 | `istio.io/waypoint-for` | Alpha | `Gateway` | Specifies what types of endpoints the waypoint will process traffic for. <br><br> Valid values: `service`, `workload`, `none` or `all`. This label is optional and the default value is `service`. |
 
 In order for your `istio.io/use-waypoint` label value to be effective, you have to ensure the waypoint is configured for the endpoint which is using it. By default waypoints accept traffic for service endpoints. For example, when you label a pod to use a specific waypoint via the `istio.io/use-waypoint` label, the waypoint should be labeled `istio.io./waypoint-for` with the value `workload` or `all`.
```
[the remark this answers is not in the document]
Comment (linsun) [reply content/en/docs/ambient/usage/add-workloads/index.md:72]: +1, referring to endpoint is super confusion

File: content/en/docs/ambient/usage/debug-waypoint/index.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
+---
+title: Debug connectivity issues with waypoints
+description: How to investigate problems routing through waypoint proxies.
+weight: 70
+owner: istio/wg-networking-maintainers
+test: no
+---
+
+This guide describes what to do if you have enrolled a workload in a waypoint proxy, but you are not seeing the expected behavior.
```
[the remark this answers is not in the document]
Comment (linsun) [reply content/en/docs/ambient/usage/debug-waypoint/index.md:9]: is the guide only for workload? I felt it is for service as well.

Comment (linsun) [inline content/en/docs/ambient/usage/debug-waypoint/index.md:9]: I recall i had some existing pre-requisites here before debugging waypoints. It is probably good to have them here.

File: content/en/docs/ambient/usage/l4-policy/index.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+The layering of {{< gloss >}}ztunnel{{< /gloss >}} and {{< gloss >}}waypoint{{< /gloss >}} proxies in Istio's ambient mode gives you a choice on whether or not you want to enable Layer 7 (L7) processing for a given workload.
+
+The Layer 4 (L4) features of Istio's [security policies](/docs/concepts/security) are supported by ztunnel, and are available in ambient mode. [Kubernetes Network Policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/) also continue to work if your cluster has a {{< gloss >}}CNI{{< /gloss >}} plugin that supports them, and can be used to provide defense-in-depth.
+
+To use L7 policies, and Istio's traffic routing features, you can [deploy a waypoint](/docs/ambient/usage/waypoint) for your workloads.
+
+## Layer 4 authorization policies
+
+The ztunnel proxy performs authorization policy enforcement when a workload is enrolled in secure overlay mode (i.e. with no waypoint proxy configured).
```
[the remark this answers is not in the document]
Comment (linsun) [reply content/en/docs/ambient/usage/l4-policy/index.md:17]: +1

File: content/en/docs/ambient/usage/l4-policy/index.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+ action: ALLOW
+ rules:
+ - from:
+   - source:
+       principals:
+       - cluster.local/ns/ambient-demo/sa/sleep
+EOF
+{{< /text >}}
+
+The behavior of the `AuthorizationPolicy` API has the same functional behavior in Istio ambient mode as in sidecar mode. When there is no `AuthorizationPolicy` provisioned, then the default action is `ALLOW`. Once a policy is provisioned, pods matching the selector in the policy only allow traffic which is explicitly allowed. In this example, pods with the label `app: httpbin` only allow traffic from sources with an identity principal of `cluster.local/ns/ambient-demo/sa/sleep`. Traffic from all other sources will be denied.
```
Comment (linsun) [inline content/en/docs/ambient/usage/l4-policy/index.md:41]: This is not entirely true.... Authzpolicy is the one that have biggest gotchas.  In this particular example (L4 authz policy), you can say it is same functional behavior.

File: content/en/docs/ambient/usage/l4-policy/index.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+   - source:
+       principals:
+       - cluster.local/ns/ambient-demo/sa/sleep
+   to:
+   - operation:
+       methods: ["GET"]
+EOF
+{{< /text >}}
+
+Even though the identity of the pod is otherwise correct, the presence of a L7 policy causes the ztunnel to deny the connection:
```
Comment (linsun) [inline content/en/docs/ambient/usage/l4-policy/index.md:72]:
```suggestion
Even though the identity of the pod is otherwise correct, the presence of a L7 policy plus the traffic isn't from waypoint causes the ztunnel to deny the connection:
```

File: content/en/docs/ambient/usage/l7-features/index.md [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+owner: istio/wg-networking-maintainers
+test: no
+---
+
+With a waypoint proxy deployed for a workload, you can enable more of [Istio's features](/docs/concepts).
+
+Ambient mode supports configuring waypoints using the Kubernetes Gateway API. Configurations that apply to a Gateway API are called _policies_.
+
+{{< warning >}}
+The Istio traffic management APIs (virtual service, destination rules etc) are not supported by the ambient data plane mode.
```
Comment (linsun) [inline content/en/docs/ambient/usage/l7-features/index.md:14]:
```suggestion
The Istio classic traffic management APIs (virtual service, destination rules etc) remains at alpha for the ambient data plane mode.
```

File: content/en/docs/ambient/usage/l7-features/index.md [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+
+## Traffic routing
+
+With a waypoint proxy deployed, you can use the following API types:
+
+|  Name  | Feature Status | Attachment |
+| --- | --- | --- |
+| `HTTPRoute` | Beta | `parentRefs` |
+| `TCPRoute` | Beta | `parentRefs` |
+| `TLSRoute` | Beta | `parentRefs` |
```
[the remark this answers is not in the document]
Comment (linsun) [reply content/en/docs/ambient/usage/l7-features/index.md:25]: +1 they should be marked as Alpha.  Not sure if we need a tip - i basically hate too many tips in docs :)

File: content/en/docs/ambient/usage/l7-features/index.md (hunk 10 above)
Comment (linsun) [reply content/en/docs/ambient/usage/l7-features/index.md:14]: Maybe just add them to the table below?

**Review by linsun — COMMENTED** [review]
LGTM overall

Made a few comments and suggestions.

**Comment by craigbox** [comment]
Thank you both!
As I have largely just rewritten or moved existing content I'm not 100% sure of what I can or should change if an explicit suggestion wasn't made. Look out for more questions on Slack, I guess.

File: content/en/docs/ambient/architecture/traffic-redirection/index.md [hunk 12]
```diff
@@ -52,7 +52,7 @@ This diagram illustrates how encrypted traffic flows between pods in the ambient
 
 ## Observing and debugging traffic redirection in ambient mode
 
-If traffic redirection is not working correctly in ambient mode, some quick checks can be made to help narrow down the problem. To demonstrate traffic redirection in action, first follow the steps described in the [ztunnel debugging guide](/docs/ambient/usage/debugging).
+If traffic redirection is not working correctly in ambient mode, some quick checks can be made to help narrow down the problem. To demonstrate traffic redirection in action, first follow the steps described in the [ztunnel debugging guide](/docs/ambient/usage/debug-ztunnel/).
```
[the remark this answers is not in the document]
Comment (craigbox) [reply content/en/docs/ambient/architecture/traffic-redirection/index.md:55]: Troubleshooting is a sensible change!

File: content/en/docs/ambient/usage/debug-waypoint/index.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (craigbox) [reply content/en/docs/ambient/usage/debug-waypoint/index.md:9]: going with "enrolled a namespace, service or workload"

File: content/en/docs/ambient/usage/debug-waypoint/index.md [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+
+To send some requests to the `reviews` `v2` pod from the `sleep` pod:
+
+{{< text bash >}}
+$ export REVIEWS_V2_POD_IP=$(kubectl get pod -l version=v2,app=reviews -o jsonpath='{.items[0].status.podIP}')
+$ kubectl exec deploy/sleep -- curl -s http://$REVIEWS_V2_POD_IP:9080/reviews/1
+{{< /text >}}
+
+Requests to the `reviews` service should be enforced by the `reviews-svc-waypoint` for any L7 policies.
+Requests to the `reviews` `v2` pod should be enforced by the `reviews-v2-pod-waypoint` for any L7 policies.
```
[the remark this answers is not in the document]
Comment (craigbox) [reply content/en/docs/ambient/usage/debug-waypoint/index.md:27]: Yes, these used to be one page.  I want this to be "how you troubleshoot your cluster", not just "how you follow a contrived bookinfo example".

I'll add some context.

File: content/en/docs/ambient/usage/debug-waypoint/index.md (hunk 3 above)
Comment (craigbox) [reply content/en/docs/ambient/usage/debug-waypoint/index.md:11]: The underlying problem is that Istio only ever used the word "policy" for security, but Gateway API uses it for "traffic routing" also.

File: content/en/docs/ambient/usage/l4-policy/index.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (craigbox) [reply content/en/docs/ambient/usage/l4-policy/index.md:17]: It's not clear to me that if you have a waypoint, and an AuthzPolicy, does ztunnel enforce the L4 parts of it, or does it ignore it and the waypoint is on the hook?  Does it matter what the policy scope is?

File: content/en/docs/ambient/usage/waypoint/index.md [hunk 14]
```diff
@@ -10,25 +10,30 @@ test: no
...
-Waypoint proxies are installed, upgraded and scaled independently from applications; an application owner should be unaware of their existence. Compared to the sidecar data plane mode, which runs an instance of the Envoy proxy alongside each workload, the number of proxies required can be substantially reduced.
+
+Waypoint proxies are installed, upgraded and scaled independently from applications; an application owner should be unaware of their existence. Compared to the sidecar {{< gloss >}}data plane{{< /gloss >}} mode, which runs an instance of the Envoy proxy alongside each workload, the number of proxies required can be substantially reduced.
 
 A waypoint, or set of waypoints, can be shared between applications with a similar security boundary. This might be all the instances of a particular workload, or all the workloads in a namespace.
 
+As opposed to {{< gloss >}}sidecar{{< /gloss >}} mode, in ambient mode policies are enforced by the **destination** waypoint. In many ways, the waypoint acts as a gateway into the namespace. Istio enforces that all traffic coming into the namespace goes through the waypoint, which then enforces all policies for that namespace.
```
[the remark this answers is not in the document]
Comment (craigbox) [reply content/en/docs/ambient/usage/waypoint/index.md:18]: Would we then be best to change the next line about "coming into the namespace" etc, as it's not just about a namespace, it can also be for a service/workload?

File: content/en/docs/ambient/usage/waypoint/index.md [hunk 15]
```diff
@@ -10,25 +10,30 @@ test: no
...
+The layered approach of ambient allows users to adopt Istio in a more incremental fashion, smoothly transitioning from no mesh, to the secure L4 overlay, to full L7 processing.
 
-Most of the features of ambient mode are provided by the ztunnel node agent.  Ztunnel is scoped to only process traffic at Layer 4 (L4), so that it can safely operate as a shared component.
+Most of the features of ambient mode are provided by the ztunnel node agent. Ztunnel is scoped to only process traffic at Layer 4 (L4), so that it can safely operate as a shared component.
 
-When you add a waypoint proxy for a workload, traffic will be forwarded from the ztunnel to that waypoint. This enables the following features:
+When you add a waypoint proxy for a workload, traffic will be forwarded from the ztunnel to that waypoint. If your applications require any of the following L7 mesh functions, you will need to use a waypoint proxy:
```
[the remark this answers is not in the document]
Comment (craigbox) [reply content/en/docs/ambient/usage/waypoint/index.md:26]: Taking the suggestion for the second half.  Was the first half ok?  Are you worried about the fact that it's not just about adding the waypoint, but the waypoint has to be the right traffic type?

File: content/en/docs/ambient/usage/waypoint/index.md (hunk 2 above)
Comment (craigbox) [reply content/en/docs/ambient/usage/waypoint/index.md:135]: We stopped saying "in the ambient mesh" but this might be a reason to bring it back.
In a world where we have sidecars and waypoints, you would then need two policies - one with selectors and one with a `targetRef`?

File: content/en/docs/ambient/usage/debug-waypoint/index.md (hunk 6 above)
Comment (craigbox) [reply content/en/docs/ambient/usage/debug-waypoint/index.md:9]: Do you remember what they were?  Are you talking about example services you need installed in order to follow these steps?

File: content/en/docs/ambient/usage/l4-policy/index.md (hunk 8 above)
Comment (craigbox) [reply content/en/docs/ambient/usage/l4-policy/index.md:41]: s/`AuthorizationPolicy` API/L4 `AuthorizationPolicy` API/

File: content/en/docs/ambient/usage/l7-features/index.md (hunk 10 above)
Comment (craigbox) [reply content/en/docs/ambient/usage/l7-features/index.md:14]: We would then have to call out the attachment type being a selector as opposed to a targetRef.  I'd love this to be in the documentation but it's too complicated to add in this PR imo.

**Comment by craigbox** [comment]
@ilrudie @linsun ready for re-review!

File: content/en/docs/ambient/usage/waypoint/index.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (ilrudie) [reply content/en/docs/ambient/usage/waypoint/index.md:26]: maybe it's just me getting too caught up in the details but waypoint-for will matter as well as how the traffic was addressed. I am working on a doc around this now though so maybe we can just move on and caveat/link to my stuff once it is ready

File: content/en/docs/ambient/usage/waypoint/index.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (ilrudie) [reply content/en/docs/ambient/usage/waypoint/index.md]: I don't think I see that line but yeah, basically anywhere we imply that a waypoint is like ingress for a namespace I'd vote to reword that.

File: content/en/docs/ambient/usage/waypoint/index.md (hunk 2 above)
Comment (ilrudie) [reply content/en/docs/ambient/usage/waypoint/index.md]: I retract my complaint. Not due to being invalid but I was simply way too in the weeds on a detail we don't support for 1.22 any way. Having said that there is potentially a weird caveat to consider for ingress gateways too. I suppose "pods enrolled in ztunnel" might help imply that ingress gateways are their own special case 🤷‍♂️

File: content/en/docs/ambient/usage/l7-features/index.md [hunk 16]
```diff
@@ -21,8 +23,12 @@ With a waypoint proxy deployed, you can use the following API types:
...
 | --- | --- | --- |
 | `HTTPRoute` | Beta | `parentRefs` |
-| `TCPRoute` | Beta | `parentRefs` |
-| `TLSRoute` | Beta | `parentRefs` |
+| `TCPRoute` | Alpha | `parentRefs` |
+| `TLSRoute` | Alpha | `parentRefs` |
+
+{{< tip >}}
+`TCPRoute` and `TLSRoute` still have Alpha status in Kubernetes Gateway API. Caution is advised if considering their adoption.
+{{< /tip >}}
```
Comment (ilrudie) [inline content/en/docs/ambient/usage/l7-features/index.md:31]: optional, I think Lin preferred no tip so remove if Lin or yourself feel strongly about it

```suggestion
```

File: content/en/docs/ambient/usage/troubleshoot-waypoint/index.md [hunk 17]
```diff
@@ -26,14 +26,14 @@ $ kubectl exec deploy/sleep -- curl -s http://$REVIEWS_V2_POD_IP:9080/reviews/1
 Requests to the `reviews` service should be enforced by the `reviews-svc-waypoint` for any L7 policies.
 Requests to the `reviews` `v2` pod should be enforced by the `reviews-v2-pod-waypoint` for any L7 policies.
 
-1.  If your L7 policy isn't enforced, run `istioctl analyze` first to check if your policy has any validation issue.
+1.  If your L7 configuration isn't applied, run `istioctl analyze` first to check if your configuration has a validation issue.
```
Comment (ilrudie) [inline content/en/docs/ambient/usage/troubleshoot-waypoint/index.md:29]: **nit** "isn't applied" sounds to me like it was rejected by validation in the Kubernetes API but that may just be me so feel free to ignore...
```suggestion
1.  If your L7 configuration isn't working as expected, run `istioctl analyze` first to check if your configuration has a validation issue.
```

File: content/en/docs/ambient/usage/troubleshoot-waypoint/index.md [hunk 18]
```diff
@@ -26,14 +26,14 @@ $ kubectl exec deploy/sleep -- curl -s http://$REVIEWS_V2_POD_IP:9080/reviews/1
...
 
     {{< text bash >}}
     $ istioctl analyze
     ✔ No validation issues found when analyzing namespace: default.
     {{< /text >}}
 
-1.  Determine which waypoint is enforcing the L7 policy for your service or pod.
+1.  Determine which waypoint is applying the L7 configuration for your service or pod.
```
Comment (ilrudie) [inline content/en/docs/ambient/usage/troubleshoot-waypoint/index.md:36]: again, sort of nit picking wording that seems off to my mind, feel free to ignore
```suggestion
1.  Determine which waypoint is implementing the L7 configuration for your service or pod.
```

File: content/en/docs/ambient/usage/waypoint/index.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (ilrudie) [reply content/en/docs/ambient/usage/waypoint/index.md:26]: well, adding to a workload would, technically speaking, require an appropriate waypoint for as well as the client to be addressing traffic to that workload instead of a service. maybe something that moves away from specifics is the call

```suggestion
When you configure redirection to a waypoint, traffic will be forwarded by ztunnel to that waypoint. If your applications require any of the following L7 mesh functions, you will need to use a waypoint proxy:
```

**Review by ilrudie — APPROVED** [review]
This looks good to me. The one bit I'm still hung up on a bit is saying things like, "adding to a workload" wrt waypoint. Maybe I'm being too pedantic and users wouldn't read it that way but alarm bells are ringing in my head because following that literally without the user knowing what they are doing may lead to a bad result. I would probably favor generic language in spots where we don't have a reason to be so specific, especially with regard to how waypoint and workloads interact.

File: content/en/docs/ambient/usage/troubleshoot-waypoint/index.md [hunk 19]
```diff
@@ -0,0 +1,107 @@
+---
+title: Troubleshoot connectivity issues with waypoints
```
Comment (linsun) [inline content/en/docs/ambient/usage/troubleshoot-waypoint/index.md:2]: Could be waypoint not enforcing policies also... that is when user would mostly need this doc for. why their waypoint isn't working, or why traffic is broken.

@ilrudie @craigbox thoughts?

**Review by linsun — APPROVED** [review]
LGTM, left a small nit

**Comment by craigbox** [comment]
@dhawton for ack; sorry, I thought Lin was a docs approver

**Review by dhawton — APPROVED** [review]
